man/halt.xml

```diff
@@ -137,12 +138,34 @@
   <refsect1>
     <title>Notes</title>
 
-    <para>These commands are implemented in a way that preserves compatibility with
+    <para>These commands are implemented in a way that preserves basic compatibility with
     the original SysV commands.
     <citerefentry><refentrytitle>systemctl</refentrytitle><manvolnum>1</manvolnum></citerefentry>
     verbs <command>halt</command>, <command>poweroff</command>,
     <command>reboot</command> provide the same functionality with some additional
     features.</para>
+
+    <para>One major difference in the behavior of these commands under
+    <literal>systemd</literal> and SysV systems
+    is that under SysV, the "<command>halt</command>" command typically caused the
+    computer to poweroff as well as halting program execution. This is because the actions
+    corresponding to <command>halt</command> and <command>poweroff</command> were both
+    implemented via <command>telinit 0</command>, i.e. forcing the system into runlevel
+    zero. Once in runlevel zero, a typical configuration would then power off the machine.
+    The <command>reboot</command> command was executed via its own runlevel, 6. Systemd
+    does a better job of differentiating the two actions "halt" and "power off",
+    by defining two targets
+    (<filename>halt.target</filename> and <filename>poweroff.target</filename>) instead
+    of a single "0" runlevel like SysV. That means it can carry the correct verb all the
+    way through to the
+    <citerefentry><refentrytitle>reboot</refentrytitle><manvolnum>2</manvolnum></citerefentry>
+    system call in <command>systemd-shutdown</command>.
+    </para>
```

Comment:
Hmmm, this explains the situation in sysvinit nicely. But after reading this, I'm want to say: why should we implement halt like this? Back in the day it made some sense because powering off didn't work properly always. But nowadays, it really should. Maybe we should stop trying to reproduce this behaviour and simple make 'halt' and alias for 'poweroff'?

Reply:
Halting is sometimes useful if you want to see/troubleshoot the final actions of the shutdown process. I would prefer if you did not remove that ability.